pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -347,7 +362,13 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	klog.Infof("Starting %s", controllerName)
 	defer klog.Infof("Shutting down %s", controllerName)
 
-	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.nodeListerSynced) {
+	cacheSynced := []cache.InformerSynced{
+		c.nodeListerSynced,
+	}
+	if c.networkConfig.IPsecConfig.AuthenticationMode == config.IPsecAuthenticationModeCert {
```

Comment:
Found `AuthenticationMode` is initialized with the config value regardless of encryption mode, then it should also check `c.networkConfig.TrafficEncryptionMode` to be more specific, otherwise there would be issue when default mode is switched to Cert or user sets "Cert" but doesn't enable encryption at all.

Reply:
fixed. Thanks!